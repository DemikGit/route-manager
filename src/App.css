.app__container {
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  /* background-color: #f5f9f9; */
  background: linear-gradient(45deg, rgba(63,151,186,1) 0%, rgba(234,83,242,1) 100%);
}
