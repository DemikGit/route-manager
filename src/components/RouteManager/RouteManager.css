.route-manager__container {
  display: flex;
  width: 100%;
  height: 100%;
  max-width: 1024px;
  margin-left: 20px;
  margin-right: 20px;
  justify-content: space-between;
}

.route-manager__controls {
  display: flex;
  flex-basis: 50%;
  flex-direction: column;
  justify-content: flex-start;
}

.route-manager__map {
  display: block;
  width: 50%;
  min-width: 300px;
  height: 100%;
}
