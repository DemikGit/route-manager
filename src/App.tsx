import * as React from 'react';
import { Component } from 'react';
import './App.css';


class App extends Component {
  public render() {
    return null;
  }
}

export default App;
