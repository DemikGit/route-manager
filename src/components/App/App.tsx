import * as React from 'react';
import { Component } from 'react';
import './App.css';

import { RouteManager } from '../RouteManager/RouteManager';

class App extends Component {
  public render() {
    return (
      <div className="app__container">
        <RouteManager text="hello" />
      </div>
    )
  }
}

export default App;
