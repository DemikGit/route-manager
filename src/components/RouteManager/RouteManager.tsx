import * as React from 'react';
import { Component } from 'react';
import { BaseInput } from '../BaseInput/BaseInput';
import './RouteManager.css'

import { GoogleMap } from "react-google-maps"
import { RMRoute } from '../../classes/RMRoute';
import { MapPreview } from '../MapPreview/MapPreview';
import { MarkerList } from '../MarkersList/MarkerList';

export interface IRouteManagerProps {
  text: string,
}
export interface IRouteManagerState {
  apiParams: {
    key: string,
    libs: string,
    version: string,
  },
  route: RMRoute,
}

export class RouteManager extends Component<IRouteManagerProps, IRouteManagerState> {
  public state = {

    apiParams: {
      key: 'AIzaSyDSnQhXn5WfyBi68hCgx5qA4pmApKkN8Jc',
      libs: 'geometry,drawing,places',
      version: '3.exp',
    },

    route: new RMRoute(),
  }

  private mapRef: React.RefObject<GoogleMap> = React.createRef();

  public componentDidMount() {
    const { route } = this.state;
    route.mapRef = this.mapRef;
    route.updateComponent = () => this.forceUpdate();
  }

  public render() {
    const { route } = this.state;
    const { key, libs, version } = this.state.apiParams;

    const url = `https://maps.googleapis.com/maps/api/js?key=${
      key }&v=${ version }&libraries=${ libs }`;

    return (
      <div className="route-manager__container">
        <div className="route-manager__controls">
          <BaseInput text="kek" onAddPoint={ route.onAddPoint } />
          <MarkerList routeMarkers={ route.markers } onPointDelete={ route.onPointDelete } />
        </div>
        <div className="route-manager__map">
          <MapPreview
            googleMapURL={ url }
            defaultCenter={{
              lat: 37.666030368995685,
              lng: -116.02545749206541,
            }}
            defaultZoom={ 15 }
            loadingElement={ <div style={{ height: '100%' }} /> }
            containerElement={
              <div className="route-manager__map__wrapper"/>
            }
            mapElement={ <div style={{ height: '100%' }} /> }
            mountMap={ this.mapRef }
          >
            { route.render() }
          </MapPreview>
        </div>
      </div>
    );
  }
}
