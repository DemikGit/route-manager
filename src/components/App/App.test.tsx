import { shallow } from 'enzyme';
import 'jest-enzyme';
import * as React from 'react';
import { RouteManager } from '../RouteManager/RouteManager';
import App from './App';

it('renders without crashing', () => {
  shallow(<App />);
});

it('have RouteManager component', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toContainReact(<RouteManager />);
  expect(wrapper).toMatchSnapshot();
});
