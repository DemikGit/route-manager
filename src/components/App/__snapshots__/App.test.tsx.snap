// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`have RouteManager component 1`] = `
<div
  className="app__container"
>
  <RouteManager />
</div>
`;
